CSharpSourceCode/Abilities/FireBallAbility.cs

```diff
@@ -24,39 +25,54 @@ public FireBallAbility() : base()
 
         protected override void OnUse(Agent casterAgent)
         {
-            if(casterAgent.IsActive() && casterAgent.Health > 0 && (casterAgent.GetMorale() > 1 || casterAgent.IsPlayerControlled) && casterAgent.IsAbilityUser())
+            if (casterAgent.IsActive() && casterAgent.Health > 0 &&
+                (casterAgent.GetMorale() > 1 || casterAgent.IsPlayerControlled) && casterAgent.IsAbilityUser())
             {
                 var scene = Mission.Current.Scene;
                 var offset = 1f;
                 var mass = 1f;
                 var lightradius = 10f;
 
                 var frame = casterAgent.LookFrame.Elevate(casterAgent.GetEyeGlobalHeight());
+                frame = TargetForAI(casterAgent, frame);
```

Comment:
maybe rename this method to better reflect its function. Something like AdjustEffectSpawnDirectionForAI.

Reply:
I went with UpdateFrameRotationForAI.

I was also thinking OverrideFrameRotationForAI, but override is also a language keyword so I figured it might be better not to.